src/drawing/DrawingThread.js

```diff
@@ -241,51 +241,49 @@ class DrawingThread extends AnnotationThread {
     }
 
     /**
-     * Overturns the last drawing stroke if it exists. Emits thenumber of undo and redo
+     * Overturns the last drawing stroke if it exists. Emits the number of undo and redo
      * actions available if an undo was executed.
      *
      * @return {void}
      */
     undo() {
         const executedUndo = this.pathContainer.undo();
         if (executedUndo) {
-            this.draw(this.drawingContext, true);
-            this.updateBoundary();
-            this.regenerateBoundary();
-
-            if (this.pathContainer.isEmpty()) {
-                this.unmountPopover();
-            } else {
-                this.renderAnnotationPopover();
-            }
-
-            this.drawBoundary();
-            this.emitAvailableActions();
+            this.updateBoundaryAndPopover();
         }
     }
 
     /**
-     * Replays the last undone drawing stroke if it exists. Emits thenumber of undo and redo
+     * Replays the last undone drawing stroke if it exists. Emits the number of undo and redo
      * actions available if a redraw was executed.
      *
      * @return {void}
      */
     redo() {
         const executedRedo = this.pathContainer.redo();
         if (executedRedo) {
-            this.draw(this.drawingContext, true);
-            this.updateBoundary();
-            this.regenerateBoundary();
+            this.updateBoundaryAndPopover();
+        }
+    }
 
-            if (this.pathContainer.isEmpty()) {
-                this.unmountPopover();
-            } else {
-                this.renderAnnotationPopover();
-            }
+    /**
+     * Updates the position of the drawing boundary and popover after an undo/redo event.
+s    *
+     * @return {void}
+     */
+    updateBoundaryAndPopover() {
+        this.draw(this.drawingContext, true);
+        this.updateBoundary();
+        this.regenerateBoundary();
+
+        this.unmountPopover();
```

Comment:
unmounting the popover will cause the position to be updated if/when it is remounted.